supervisor/host/network.py

```diff
@@ -283,6 +283,15 @@ async def apply_changes(
 
         # Create new interface (like vlan)
         elif not inet:
+            # For VLAN interfaces, check if one already exists with same ID on same parent
+            if interface.type == InterfaceType.VLAN and interface.vlan:
+                vlan_interface_name = f"{interface.vlan.interface}.{interface.vlan.id}"
+                if vlan_interface_name in self.sys_dbus.network:
+                    raise HostNetworkError(
+                        f"VLAN {interface.vlan.id} already exists on interface {interface.vlan.interface}",
+                        _LOGGER.error,
+                    )
```

Comment:
Since `interface.name` is always empty for vlans, won't it go into this code path any time you are adding or updating a vlan interface? Currently I only see a test case for adding a new vlan in `tests/api/test_network.py`. I would recommend adding one for updating the details of an existing vlan to make sure this doesn't break that.

And if update does go down a separate code path somehow then a test case for the scenario you're trying to prevent (creating duplicate vlans) would be good. But I think the comment is misleading and the reason we don't have this check is to allow updates of existing vlan interfaces we've created.

Reply:
Updating a VLAN is handled by regular `ha network update` command, just using the interface with VLAN id (as in `<interface>.<vlanid>`). I think this code path is really only used to create VLAN, and only that.